MapboxCoreNavigation/LegacyRouteController.swift

```diff
@@ -373,7 +375,6 @@ open class LegacyRouteController: NSObject, Router, InternalRouter, CLLocationMa
             }
 
```

Comment:
@Woildan semantically, the solution should go right about the guard for checking the value of `route`. Right above [line #375](https://github.com/mapbox/mapbox-navigation-ios/pull/2093/files#diff-ac5494b213aa895d04f334c942cbf29cL375). We want to respect the error response returned after the `getDirections` call.

Solution should look like this:
```swift
strongSelf.isRerouting = false
guard let route = route else { return }
```

Reply:
@Woildan I understand your concern. Well, since the `isRerouting` isn't getting updated, in that case when an error occurs, we should set that value on [line # 366](https://github.com/mapbox/mapbox-navigation-ios/pull/2093/files#diff-ac5494b213aa895d04f334c942cbf29cL366) before inspecting the `error`. That addresses the `route` issue... would you agree?

Reply:
Okay I've made the change.